tools/deb/istio-iptables.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# Initialization script responsible for setting up port forwarding for Istio sidecar.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+  echo "${0} -p ENVOY_PORT -u UID -s IN_SERVICE_PORTS [-h]"
+  echo ''
+  echo '  -p: Specify the envoy port to which redirect all TCP traffic'
+  echo '  -u: Comma separate list of UIDs for which the redirection is not'
+  echo '      applied. Typically, this is the UID of the proxy container'
+  echo '  -i: Comma separated list of IP ranges in CIDR form to redirect to envoy (optional)'
+  echo '  -s: Comma separated list of incoming service IPs to intercept (optional)'
+  echo ''
+}
+
+IP_RANGES_INCLUDE=""
+
+while getopts ":p:u:e:i:h:s" opt; do
+  case ${opt} in
+    p)
+      ENVOY_PORT=${OPTARG}
+      ;;
+    u)
+      ENVOY_UID=${OPTARG}
+      ;;
+    i)
+      IP_RANGES_INCLUDE=${OPTARG}
+      ;;
+    s)
+      ISTIO_LOCAL_PORTS=${OPTARG}
+      ;;
+    h)
+      usage
+      exit 0
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "${ENVOY_PORT-}" ]]; then
+  ENVOY_PORT=15001
+fi
+
+if [[ -z "${ENVOY_UID-}" ]]; then
+  ENVOY_UID=`id -u ${ENVOY_USER-istio}`
+  ENVOY_UID=${ENVOY_UID},0
+fi
+
+if [ "$1" = "clean" ]; then
+  iptables -F -t nat
+  iptables -t nat -X ISTIO_REDIRECT
+  iptables -t nat -X ISTIO_OUTPUT
+  exit 0
+fi
+
+
+# Create a new chain for redirecting inbound traffic to the common Envoy port.
+iptables -t nat -X ISTIO_REDIRECT
+iptables -t nat -N ISTIO_REDIRECT                                             -m comment --comment "istio/redirect-common-chain"
+iptables -t nat -A ISTIO_REDIRECT -p tcp -j REDIRECT --to-port ${ENVOY_PORT}  -m comment --comment "istio/redirect-to-envoy-port"
+
+IFS=,
+if [ "${ISTIO_LOCAL_PORTS}" == "*" ]; then
```

Comment:
This assumes a single local address/interface. Is that okay for 0.3, or do we need to support multiple address/port tuples for inbound redirection?